astropy/io/ascii/ui.py

```diff
@@ -666,6 +668,26 @@ def write(table, output=None,  format=None, Writer=None, fast_writer=True, **kwa
 
     table = Table(table, names=kwargs.get('names'))
 
+    table0 = table[:0].copy()
+    core._apply_include_exclude_names(table0, kwargs.get('names'),
+                    kwargs.get('include_names'), kwargs.get('exclude_names'))
+    diff_format_with_names = set(kwargs.get('formats', [])) - set(table0.colnames)
+    same_include_names_with_exclude_names = set(kwargs.get('include_names', []))
```

Comment:
Please remove these two (`same_include_...` and `diff_include_...`) and the `if same_include_..` below.  The original issue was to validate `formats` given any set of `names`, `include_names`, `exclude_names` inputs.  We do not require separate consistency of `include_names` and `exclude_names` since they do have a well-defined precedence order so overlapping is OK.